native/cocos/core/scene-graph/Node.cpp

```diff
@@ -823,7 +827,7 @@ void Node::setWorldRotationFromEuler(float x, float y, float z) {
     Quaternion::fromEuler(x, y, z, &_worldRotation);
     if (_parent) {
         _parent->updateWorldTransform();
-        _localRotation = _parent->getWorldRotation().getConjugated() * _worldRotation;
+        _localRotation = _parent->_worldRotation.getConjugated() * _worldRotation;
```

Comment:
`_parent->updateWorldTransform();` is invoked at the previous line. So no need to invoke getWorldRotation again since the world transform has already been updated.

```c++
const Quaternion &Node::getWorldRotation() const { //NOLINT(misc-no-recursion)
    const_cast<Node *>(this)->updateWorldTransform();
    return _worldRotation;
}
```